### PR TITLE
Edit this page + ToC design

### PR DESCRIPTION
**Pull request by zbayoff**
(Reopened from https://github.com/DataDog/documentation/pull/5655)

1. Adds `Edit` button in TOC which will link to the source `md` file in github, if it exists. If there is no `md` file, as would be the case for integrations, the `Edit` button links to open a new issue in Github. 
2. Updates the Mobile and Desktop TOC design. The TOC should now always be visible on desktop (if not disabled), and will be toggleable on mobile.
3. The `.mainContent-wrapper` column is reduced from the current 9 columns to 7 columns for better readability. The 12 column layout is now 3 (sidenav) - 7 (main Content) - 2 (TOC). There may be a couple layout issues regarding tabs fitting into the smaller column width. 

https://trello.com/c/XbTcdAlX/3367-add-dependencies-in-all-public-documentation-page

https://docs-staging.datadoghq.com/zach/editPage/

### CODE REVIEW

**Comment by zbayoff** [comment]
https://github.com/DataDog/documentation/pull/5655#issuecomment-564928914

**Comment by ruthnaebeck** [comment]
It should be possible to disable the TOC and have pages stretch for the [main integration page](https://docs-staging.datadoghq.com/zach/editPage/integrations) and landing pages ([example](
https://docs-staging.datadoghq.com/zach/editPage/getting_started)).

For the integrations located in dogweb we should direct people to `https://github.com/DataDog/documentation/issues/new?template=issue.md` instead of `https://github.com/DataDog/documentation/issues/new/choose`. Also, can we check if there's a way to populate the referring URL into the GitHub text box?

Is there a way to handle pages that get combined ([example](https://docs-staging.datadoghq.com/zach/editPage/integrations/activemq/#activemq-xml-integration))?

**Comment by ruthnaebeck** [comment]
LGTM 🙂

File: layouts/partials/page-edit.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                {{ end }}
+                
+        {{ end }}
+
+    {{ end }}
+
+{{ else if (eq $ctx.Section "integrations" )}}
+{{/* Files with integrations path and no dependencies link to new issue */}}
+
+    {{ $ctx.Scratch.Set "editURL" "https://github.com/DataDog/documentation/issues/new?template=issue.md" }}
```
Comment (l0k0ms) [inline layouts/partials/page-edit.html]: This needs to be merged after: https://github.com/DataDog/documentation/pull/6471

```suggestion
    {{ $ctx.Scratch.Set "editURL" "https://github.com/DataDog/documentation/issues/new?template=edit_this_page.md" }}
```

**Pull request merged**